airbyte-webapp/src/packages/cloud/services/auth/AuthService.tsx

```diff
@@ -133,6 +135,26 @@ export const AuthenticationProvider: React.FC = ({ children }) => {
       async confirmPasswordReset(code: string, newPassword: string): Promise<void> {
         await authService.finishResetPassword(code, newPassword);
       },
+      async signUpWithEmailLink({ name, email, password, news }): Promise<void> {
+        let firebaseUser: FbUser;
+
+        try {
+          ({ user: firebaseUser } = await authService.signInWithEmailLink(email));
+          await authService.updatePassword(password);
+        } catch (e) {
+          await authService.signOut();
+          if (e.message === EmailLinkErrorCodes.LINK_EXPIRED) {
+            await userService.resendWithSignInLink({ email });
+          }
+          throw e;
+        }
+
+        if (firebaseUser) {
+          const user = await userService.getByAuthId(firebaseUser.uid, AuthProviders.GoogleIdentityPlatform);
+          await userService.update({ userId: user.userId, authUserId: firebaseUser.uid, name, news });
+          await onAfterAuth(firebaseUser, { ...user, name });
+        }
```

Comment:
Since we're throwing unconditionally in the `catch` above, we should never reach this in case an exception is thrown, thus we should be able to move this to line 144 into the `try` and thus could just have a `const firebaseUser` instead of the `let` and also would save that `if` statement.

Reply:
The reason I did it this as a failsafe if userService fails for some reason. This makes it so that the user will be created even if the name could not be set.

Reply:
Sorry, I think I put this confusing, let me try to rephrase. If the `userService` would fail the above `catch` would trigger and would at the end of it always rethrow the error (line 149), thus in case any of those APIs fail we'll not hit this code anyway. I think it makes only sense to put this outside the `try` if we'd remove line 149 and not throw e in all error cases again (thus might break other things in this case).